UIAutomationTests/TestFixture.cs

```diff
@@ -0,0 +1,218 @@
+        string originalQueryPqStr;
+
+        Application app;
+        UIA3Automation automation;
+
+        ConditionFactory cf;
+
+        AutomationElement mainWindow;
+
+        AutomationElement debugTargetButtonAE;
+        Button debugTargetButton;
+        AutomationElement MQueryOutput;
+        AutomationElement[] tabItemAEs;
+
+        //https://stackoverflow.com/questions/12976319/xunit-net-global-setup-teardown
+        //Do "global" initialization here; Only called once.
+        public TestFixture()
+        {
+            PrepVisualStudio();
+
+            SetupConditionFactory();
+
+            AcquireDebugTargetButton();
+            PressDebugTargetButton();
+
+            AcquireMQueryWindowAndAcquireTabsWhenFullyLoaded();
+
+            //entering the credentials seems to be more reliable
+            EnterCredentials();
+
+            SaveAndBackupOriginalQueryString();
+        }
+
+        private void SetupConditionFactory()
+        {
+            cf = new ConditionFactory(new UIA3PropertyLibrary());
+        }
+
+        private void EnterCredentials()
+        {
+            //the errors tab will pop up and ask for credentials
+            var errorsTabAE = tabItemAEs[2];
+            errorsTabAE.AsTabItem().Select();
+
+            var comboBoxes = WaitUntilMultipleFoundAsync(errorsTabAE, FlaUI.Core.Definitions.TreeScope.Descendants, cf.ByControlType(FlaUI.Core.Definitions.ControlType.ComboBox)).Result;
+            //comboBoxes = errorsTabAE.FindAll(FlaUI.Core.Definitions.TreeScope.Descendants, cf.ByControlType(FlaUI.Core.Definitions.ControlType.ComboBox));
+            var cbCredentialType = comboBoxes[1].AsComboBox();
+            cbCredentialType.Select(0);
+
+            var textBoxes = errorsTabAE.FindAll(FlaUI.Core.Definitions.TreeScope.Descendants, cf.ByControlType(FlaUI.Core.Definitions.ControlType.Edit));
+            textBoxes[0].AsTextBox().Text = "sys";
+            textBoxes[1].AsTextBox().Text = "exasol";
+
+            var buttons = errorsTabAE.FindAll(FlaUI.Core.Definitions.TreeScope.Descendants, cf.ByControlType(FlaUI.Core.Definitions.ControlType.Button));
+            buttons[0].AsButton().Invoke();
+
+        }
+
+        private void PrepVisualStudio()
+        {
+            app = FlaUI.Core.Application.Launch(vsExecutablePath, slnPath);
+
+            automation = new UIA3Automation();
+            automation.ConnectionTimeout = new TimeSpan(0, 1, 0);
+            automation.TransactionTimeout = new TimeSpan(0, 1, 0);
+
+            //https://markheath.net/post/async-antipatterns
+            mainWindow = WaitUntillSlnIsLoadedAsync(app, automation).Result;
+
+        }
+
+        private void AcquireMQueryWindowAndAcquireTabsWhenFullyLoaded()
+        {
+            MQueryOutput = WaitUntilFindFirstFoundAsync(mainWindow, FlaUI.Core.Definitions.TreeScope.Descendants, cf.ByName("M Query Output")).Result;
+
+            //acquire tabs
```

Comment:
same comment as above. Usually the same is true for comments in your code. You can document your code by using the same level of abstraction from the POV of the user.
For example, here you can create an `acquireTabs` method instead. It will improve the readability and make it easier to understand and maintain, and will make the code a less clutter.

Reply:
I removed some of the excessive whitespace.